tests/Fake/DataObjectFake.php

```diff
@@ -1,52 +1,49 @@
 <?php
 
-
 namespace SilverStripe\SearchService\Tests\Fake;
 
 use SilverStripe\Dev\TestOnly;
 use SilverStripe\ORM\DataObject;
 use SilverStripe\SearchService\Extensions\SearchServiceExtension;
 use SilverStripe\Security\Member;
+use stdClass;
 
 class DataObjectFake extends DataObject implements TestOnly
 {
-    public $can_view;
 
-    private static $db = [
+    private static array $db = [
         'Title' => 'Varchar',
         'ShowInSearch' => 'Boolean',
-        'Sort' => 'Int'
+        'Sort' => 'Int',
     ];
 
-    private static $casting = [
+    private static array $casting = [
         'getDBHTMLText' => 'HTMLText',
     ];
 
-    private static $many_many = [
+    private static array $many_many = [
         'Tags' => TagFake::class,
     ];
 
-    private static $has_many = [
+    private static array $has_many = [
         'Images' => ImageFake::class,
     ];
 
-    private static $has_one = [
+    private static array $has_one = [
         'Member' => Member::class,
     ];
 
-    private static $default_sort = 'Sort ASC';
+    private static string $default_sort = 'Sort ASC';
+
+    private static string $table_name = 'DataObjectFake';
 
-    private static $extensions = [
+    private static array $extensions = [
         SearchServiceExtension::class,
     ];
 
-    public function canView($member = null)
+    public function canView($member = null) // phpcs:ignore SlevomatCodingStandard.TypeHints
     {
-        if (is_callable($this->can_view)) {
-            $func = $this->can_view;
-            return $func();
-        }
-        return $this->can_view;
```

Comment:
Opted to make new DataObjects that returned `false` in their `canView()`, rather than relying on this sort of thing.